### PR TITLE
Support TransposeConv op

### PR DESCRIPTION
**Pull request by Honry**

### CODE REVIEW

**Comment by Honry** [comment]
@huningxin, PTAL, thanks!

File: tensorflow/lite/delegates/webnn/webnn_delegate.cc [hunk 1]
```diff
@@ -2743,6 +2758,146 @@ class Subgraph {
...
+
+    std::string auto_pad;
+    TF_LITE_ENSURE_STATUS(CalculatePadding(
+        logging_context, deconv_params->padding, auto_pad, node_index));
+
+    if (detect_supported_op) {
+      TF_LITE_ENSURE_STATUS(CheckWebNNOpSupport(builder, "convTranspose2d"));
+    } else {
+      emscripten::val options = emscripten::val::object();
+      options.set("autoPad", emscripten::val(auto_pad));
```
Comment (huningxin) [inline tensorflow/lite/delegates/webnn/webnn_delegate.cc:2873]: If the `autoPad` is `explicit`, should the `padding` be computed? Refer to XNNPACK delegate [`CalculateTransposeConvPaddings()`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/xnnpack/xnnpack_delegate.cc#L5700).

Comment (Honry) [reply tensorflow/lite/delegates/webnn/webnn_delegate.cc:2873]: Looks into the [CalculateTransposeConvPaddings()](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/xnnpack/xnnpack_delegate.cc#L1148), when the `autoPad` is `explicit` (i.e. `kTfLitePaddingValid`),  all the padding value will be set to 0.

`*padding_top = *padding_bottom = *padding_left = *padding_right = 0;`

That's the default value of WebNN's `padding`, so we can just ignore it.

Comment (huningxin) [reply tensorflow/lite/delegates/webnn/webnn_delegate.cc:2873]: Thanks for the details. Let's ignore it for now.

**Pull request merged**